src/hippopt/robot_planning/expressions/centroidal.py

```diff
@@ -0,0 +1,92 @@
+import casadi as cs
+
+
+def centroidal_dynamics_with_point_forces(
+    number_of_points: int,
+    assume_unitary_mass: bool = False,
+    mass_name: str = "m",
+    gravity_name: str = "g",
+    com_name: str = "com",
+    point_position_names: list[str] = None,
+    point_force_names: list[str] = None,
+    options: dict = None,
+    **_,
+) -> cs.Function:
+    options = {} if options is None else options
+
+    if point_position_names is None:
+        point_position_names = []
+        for i in range(number_of_points):
+            point_position_names.append(f"p{i}")
+
+    assert len(point_position_names) == number_of_points
+
+    if point_force_names is None:
+        point_force_names = []
+        for i in range(number_of_points):
+            point_force_names.append(f"f{i}")
+
+    assert len(point_force_names) == number_of_points
+
+    input_vars = []
+
+    if assume_unitary_mass:
+        m = 1.0
+    else:
+        m = cs.MX.sym(mass_name, 1)
+        input_vars.append(m)
+
+    g = cs.MX.sym(gravity_name, 6)
+    input_vars.append(g)
+
+    x = cs.MX.sym(com_name, 3)
+    input_vars.append(x)
+
+    p = []
+    f = []
+    for i in range(number_of_points):
+        p.append(cs.MX.sym(point_position_names[i], 3))
+        input_vars.append(p[i])
+        f.append(cs.MX.sym(point_force_names[i], 3))
+        input_vars.append(f[i])
+
+    input_names = []
+    for var in input_vars:
+        input_names.append(var.name())
+
+    h_g = m @ g
+
+    for i in range(number_of_points):
+        h_g = h_g + cs.vertcat(f[i], cs.cross(p[i] - x, f[i]))
```

Comment:
I had to edit this a bit since ``sum``  does not exist, but I managed to find a one-command solution. Done in https://github.com/ami-iit/hippopt/pull/10/commits/18b27a625c8ccbf7df07460339c7d195a5d683f6